google-account-plugin/src/com/google/cloud/tools/intellij/AccountPluginInitializationComponent.java

```diff
@@ -53,9 +56,26 @@ public void initComponent() {
     }
     if (!ApplicationManager.getApplication().isUnitTestMode()) {
       configureUsageTracking();
+      initUsageTracker();
     }
   }
 
+  @VisibleForTesting
+  void initUsageTracker() {
+    String platformPrefix = System.getProperty("idea.platform.prefix");
+    String trackingId = new PropertiesFilePluginFlags().getAnalyticsId();
+
+    Predicate<Object> userOptedIn = new Predicate<Object>() {
+      @Override
+      public boolean apply(@javax.annotation.Nullable Object object) {
+        return UsageTrackerManager.getInstance().hasUserOptedIn();
+      }
+    };
+
+    IdeaUsageTracker usageTracker = ServiceManager.getService(IdeaUsageTracker.class);
+    usageTracker.init(platformPrefix, trackingId, userOptedIn);
```

Comment:
Oh so basically you're using the Predicate as a way of dynamically getting the opt in status. Why not just call `UsageTrackerManager.getInstance().hasUserOptedIn()` each time its needed? This seems like a convoluted use of Predicate unless I still don't understand?